circle.yml

```diff
@@ -21,19 +24,26 @@ machine:
     PATH: "$HOME/.config/composer/vendor/bin:$PATH"
   hosts:
     local.mysiteurl: 127.0.0.1
+  services:
+    - docker
 
 dependencies:
   cache_directories:
     - ~/.composer/cache
     # Cache DB dumps directory.
     - .data
   pre:
+    # Stopping default mysql service so the db container can expose itself on port 3306.
+    - sudo service mysql stop
     # Download DB dump from public URL to test Drupal-dev project.
     # Remove the line below in your project.
-    - if [ ! -f $BUILD_DIR/.data/db.dist.sql ] ; then mkdir $BUILD_DIR/.data && curl -L https://goo.gl/WFtJbT -o $BUILD_DIR/.data/db.dist.sql; fi
+    - if [ ! -f "${DB_CONTAINER_TAR}" ] ; then mkdir $BUILD_DIR/.data && curl -L https://goo.gl/WFtJbT -o $DB_SOURCE_DUMP; bash <(curl -s "${DB_CONTAINER_SCRIPT}"); fi
     # [META] Uncomment the line below to download DB dump from FTP, using
     # variables set in Circle CI UI.
-    # - if [ ! -f $BUILD_DIR/.data/db.dist.sql ] ; then mkdir $BUILD_DIR/.data && curl -u $FTP_USER:$FTP_PASS "ftp://$FTP_HOST/db_d7.dist.sql" -o $BUILD_DIR/.data/db.dist.sql; fi
+    # - if [ ! -f "${DB_CONTAINER_TAR}" ] ; then mkdir $BUILD_DIR/.data && curl -u $FTP_USER:$FTP_PASS "ftp://$FTP_HOST/db_d7.dist.sql" -o $DB_SOURCE_DUMP; bash <(curl -s "${DB_CONTAINER_SCRIPT}"); fi
+    # Load db container image from cache directory.
+    - docker load --input "${DB_CONTAINER_TAR}"
```

Comment:
same as above - multiline YAML and group with previous so that there is a single point of modification